### PR TITLE
fix: avoid unexpected changes to searchHooks

### PR DESCRIPTION
**Pull request by Timeless0911**
## Summary

Previously in `siteDataVMPlugin` we modify the siteData object to deal with some issues?

But it's a shallow copy, the config will be used in other runtime module like `searchHookVMPlugin`.

In below codes, searchHooks has been set undefined, which cause `searchHookVMPlugin` inject an empty module.

https://github.com/web-infra-dev/rspress/blob/8b0649b731f5751947e70be0c8245d1f92dd5bf6/packages/core/src/node/runtimeModule/siteData/index.ts#L31

https://github.com/web-infra-dev/rspress/blob/8b0649b731f5751947e70be0c8245d1f92dd5bf6/packages/core/src/node/runtimeModule/siteData/index.ts#L105

https://github.com/web-infra-dev/rspress/blob/8b0649b731f5751947e70be0c8245d1f92dd5bf6/packages/core/src/node/runtimeModule/siteData/index.ts#L120-L122

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

**Review by 10Derozan — DISMISSED** [review]

File: packages/core/src/node/runtimeModule/siteData/index.ts [hunk 1]
```diff
@@ -28,7 +28,8 @@ function deletePriviteKey<T>(obj: T): T {
 
 export async function siteDataVMPlugin(context: FactoryContext) {
   const { config, alias, userDocRoot, routeService, pluginDriver } = context;
-  const userConfig = config;
+  // deep clone to avoid unexpected changes to objects
+  const userConfig = JSON.parse(JSON.stringify(config));
```
Comment (chenjiahan) [inline packages/core/src/node/runtimeModule/siteData/index.ts:32]: Will the config contain some properties that can not be serialized?

Comment (chenjiahan) [reply packages/core/src/node/runtimeModule/siteData/index.ts:32]: such as RegExp or function

Comment (Timeless0911) [reply packages/core/src/node/runtimeModule/siteData/index.ts]: see the latest code, just use Object.assign to prevent the search object being modified

Comment (chenjiahan) [reply packages/core/src/node/runtimeModule/siteData/index.ts]: LGTM

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**